lib/tapioca/gem/listeners/not_owned_constants.rb

```diff
@@ -0,0 +1,32 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Tapioca
+  module Gem
+    module Listeners
+      class NotOwnedConstants < Base
```

Comment:
I am still not a fan of this name, tbh. What do you think of:
- `ForeignConstant` as the name of a constant that we discover that we need to handle but wasn't declared by the current gem.
- `ForeignConstantFound` as the event name